File: src/main/java/src/SistemaGestionEstudiante/Estudiante.java
package SistemaGestionEstudiante;

import java.util.ArrayList;
import java.util.List;

public class Estudiante {
    private String nombre;
    private List<Double> calificaciones;

    public Estudiante(String nombre) {
        this.nombre = nombre;
        this.calificaciones = new ArrayList<>();
    }

    public String getNombre() {
        return nombre;
    }

    public void agregarCalificacion(double calificacion) {
        calificaciones.add(calificacion);
    }

    public double calcularPromedio() {
        if (calificaciones.isEmpty()) {
            return 0.0;
        }
        double suma = 0;
        for (double calificacion : calificaciones) {
            suma += calificacion;
        }
        return suma / calificaciones.size();
    }

    @Override
    public String toString() {
        return "Estudiante{" +
                "nombre='" + nombre + '\'' +
                ", calificaciones=" + calificaciones +
                ", promedio =" + calcularPromedio() +
                '}';
    }
}

File: src/main/kotlin/SistemaGestionEstudiante/Estudiante.kt
package org.example.SistemaGestionEstudiante

class Estudiante(val nombre: String) {
    private val calificaciones: MutableList<Double> = ArrayList()

    fun agregarCalificacion(calificacion: Double) {
        calificaciones.add(calificacion)
    }

    fun calcularPromedio(): Double {
        if (calificaciones.isEmpty()) {
            return 0.0
        }
        var suma = 0.0
        for (calificacion in calificaciones) {
            suma += calificacion
        }
        return suma / calificaciones.size
    }

    override fun toString(): String {
        return "Estudiante{" +
                "nombre='" + nombre + '\'' +
                ", calificaciones=" + calificaciones +
                ", promedio = " + calcularPromedio() +
                '}'
    }
}